usr/lib/pkcs11/ep11_stdll/ep11_specific.c

```diff
@@ -5816,14 +5816,31 @@ CK_RV ep11tok_get_mechanism_info(STDLL_TokData_t * tokdata,
         pInfo->ulMinKeySize = 1024;
         break;
 
+    case CKM_SHA_1_HMAC:
+    case CKM_SHA_1_HMAC_GENERAL:
     case CKM_SHA224_HMAC:
+    case CKM_SHA224_HMAC_GENERAL:
     case CKM_SHA256_HMAC:
+    case CKM_SHA256_HMAC_GENERAL:
     case CKM_SHA384_HMAC:
+    case CKM_SHA384_HMAC_GENERAL:
     case CKM_SHA512_HMAC:
+    case CKM_SHA512_HMAC_GENERAL:
+    case CKM_SHA512_224_HMAC:
+    case CKM_SHA512_224_HMAC_GENERAL:
+    case CKM_SHA512_256_HMAC:
+    case CKM_SHA512_256_HMAC_GENERAL:
+        /*
+         * EP11 may report ulMinKeySize in bytes (but ulMaxKeySize in bits).
+         * Adjust this if ulMinKeySize appears to be way too small.
+         */
+        if (pInfo->ulMinKeySize < 80)
```

Comment:
At this time min is always in bytes, no? If this changes in future ep11 versions, we should check the version number instead of making an assumptions here (that everything < 80 is in bytes).

Reply:
Agreed, but as far as I know there currently is no easy way to determine the EP11 host library version at runtime....

Currently min is always reported in BYTES, and EP11 does not support keys < 80 bits. So the min value in bytes can not be less then 10 bytes (or 80 bits).  So when EP11 reports min as BYTEs this check will turn it into BITS. Once EP11 reports min as BITS, the check will leave min as is (i.e. as BITS), because min then is >= 80.

We could even add another check for max the same way.  Currently max is reported as BITS, so the check will leave max as is (i.e. in BITS). If EP11 decides in the future to report min and max both as BYTES, then the check will turn max also to BITS, which is what we want to report externally.

Reply:
Lets handle units in our code as they are now i.e. min is interpreted as bytes, max as bits. If units change with future ep11 version they must offer a way to detect it e.g. provide version information so we can just wrap our code with "if (old_ep11_version) .. else ..".